oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -340,6 +343,23 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), content);
     request.setParser(new JsonObjectParser(jsonFactory));
 
+    request.setIOExceptionHandler(new HttpBackOffIOExceptionHandler(new ExponentialBackOff()));
+    request.setUnsuccessfulResponseHandler(
```

Comment:
This is just a result of how the built-in error handlers for google-http-java-client are organized. HTTP errors will ultimately result in an `HttpResponseException`, which is a subclass of `IOException`, but the library has a separate handler for retrying HTTP errors, which is convenient since we do not retry all HTTP error codes. It is true that the original bug report concerned a `SocketTimeoutException`, which would be handled by the `IOExceptionHandler` alone, but it seems that the same sort of catastrophic failure could occur in the case of HTTP errors as well.